packages/util-waiter/src/poller.ts

```diff
@@ -24,6 +25,11 @@ export const runPolling = async <Client, Input>(
   input: Input,
   acceptorChecks: (client: Client, input: Input) => Promise<WaiterResult>
 ): Promise<WaiterResult> => {
+  const { state } = await acceptorChecks(client, input);
+  if (state !== WaiterState.RETRY) {
+    return { state };
+  }
```

Comment:
nit: move to a function so that it can be called from both outside and inside the loop.